doc/source/whatsnew/v1.3.0.rst

```diff
@@ -965,6 +965,7 @@ Other
 - Bug in :meth:`DataFrame.equals`, :meth:`Series.equals`, :meth:`Index.equals` with object-dtype containing ``np.datetime64("NaT")`` or ``np.timedelta64("NaT")`` (:issue:`39650`)
 - Bug in :func:`pandas.util.show_versions` where console JSON output was not proper JSON (:issue:`39701`)
 - Bug in :meth:`DataFrame.convert_dtypes` incorrectly raised ValueError when called on an empty DataFrame (:issue:`40393`)
+- Bug in :meth:`DataFrame.agg()` not sorting the non-concatenated axis in the order of the provided aggragation functions when one or more aggregation function fails to produce results (:issue:`33634`).
```

Comment:
It seems to me "the non-concatenated axis" is really only meaningful to those who understand the internals. I think this description makes sense when that phrase is removed.

Also, no period at the end.

Reply:
Adjusted